--- wtk/src/org/apache/pivot/wtk/skin/LabelSkin.java	(revision 1185388)
+++ wtk/src/org/apache/pivot/wtk/skin/LabelSkin.java	(working copy)
@@ -19,7 +19,9 @@
 import java.awt.BasicStroke;
 import java.awt.Color;
 import java.awt.Font;
+import java.awt.Graphics;
 import java.awt.Graphics2D;
+import java.awt.PrintGraphics;
 import java.awt.Transparency;
 import java.awt.font.FontRenderContext;
 import java.awt.font.GlyphVector;
@@ -343,7 +345,20 @@
                     }
                 }
 
-                graphics.drawGlyphVector(glyphVector, x, y + ascent);
+//                graphics.drawGlyphVector(glyphVector, x, y + ascent);
+                // workaround for PIVOT-511
+                Graphics g = (Graphics) graphics;
+                if (g instanceof PrintGraphics) {
+                  Label label = (Label)getComponent();
+                  String text = label.getText();
+                  if (text != null && text.length() > 0) {
+                      graphics.drawString(text, x, y + ascent);
+                  }
+                } 
+                else {
+                  graphics.drawGlyphVector(glyphVector, x, y + ascent);
+                }
+
 
                 // Draw the text decoration
                 if (textDecoration != null) {
